boot/bootutil/src/loader.c

```diff
@@ -950,6 +950,76 @@ boot_update_security_counter(uint8_t image_index, int slot,
 }
 #endif /* MCUBOOT_HW_ROLLBACK_PROT */
 
+#if defined(CONFIG_MCUBOOT_CLEANUP_UNUSABLE_SECONDARY) &&\
+defined(PM_S1_ADDRESS) || defined(CONFIG_SOC_NRF5340_CPUAPP)
```

Comment:
Do we really need this `#if` section? This issue can be reproduced also on other dks (e.g. nrf52) when upload image without b0.

Reply:
-> when upload image built for configuration without b0.

good point

Reply:
@gchwier This code is reachable for nRF52 as well if it supports PM_S1_ADDRESS.